Assets/RGLUnityPlugin/Scripts/RadarSensor.cs

```diff
@@ -0,0 +1,223 @@
+
+        public void Start()
+        {
+            sceneManager = FindObjectOfType<SceneManager>();
+            if (sceneManager == null)
+            {
+                // TODO(prybicki): this is too tedious, implement automatic instantiation of RGL Scene Manager
+                Debug.LogError($"RGL Scene Manager is not present on the scene. Destroying {name}.");
+                Destroy(this);
+                return;
+            }
+            OnValidate();
+
+            // Apply initial transform of the sensor.
+            lastTransform = gameObject.transform.localToWorldMatrix;
+        }
+
+        public void OnValidate()
+        {
+            // This tricky code ensures that configuring from a preset dropdown
+            // in Unity Inspector works well in prefab edit mode and regular edit mode.
+            bool presetChanged = validatedPreset != modelPreset;
+            bool firstValidation = validatedPreset == null;
+            if (!firstValidation && presetChanged)
+            {
+                configuration = RadarConfigurationLibrary.ByModel[modelPreset];
+            }
+            ApplyConfiguration(configuration);
+            validatedPreset = modelPreset;
+        }
+
+        private void ApplyConfiguration(RadarConfiguration newConfig)
+        {
+            if (rglGraphRadar == null)
+            {
+                return;
+            }
+
+            rglGraphRadar.UpdateNodeRaysFromMat3x4f(RadarRaysNodeId, newConfig.GetRayPoses())
+                .UpdateNodeRaysSetRange(RadarRangeNodeId, newConfig.GetRayRanges())
+                .UpdateNodePointsRadarPostprocess(RadarPostprocessNodeId, newConfig.rangeSeparation, newConfig.azimuthSeparation * Mathf.Deg2Rad)
+                .UpdateNodeGaussianNoiseAngularRay(NoiseRaysNodeId,
+                    newConfig.noiseParams.angularNoiseMean * Mathf.Deg2Rad,
+                    newConfig.noiseParams.angularNoiseStDev * Mathf.Deg2Rad)
+                .UpdateNodeGaussianNoiseDistance(NoiseDistanceNodeId, newConfig.noiseParams.distanceNoiseMean,
+                    newConfig.noiseParams.distanceNoiseStDevBase, newConfig.noiseParams.distanceNoiseStDevRisePerMeter);
+
+            rglGraphRadar.SetActive(NoiseDistanceNodeId, applyDistanceGaussianNoise);
+            rglGraphRadar.SetActive(NoiseRaysNodeId, applyAngularGaussianNoise);
+        }
+
+        // TODO(msz-rai): Radars should be triggered together with LiDARs to achieve the maximum performance.
+        // TODO(msz-rai): See FixedUpdate method in LidarSensor component for more details.
+        private void FixedUpdate()
+        {
+            if (lastUpdateFrame != Time.frameCount)
+            {
+                fixedUpdatesInCurrentFrame = 0;
+                lastUpdateFrame = Time.frameCount;
+            }
+            fixedUpdatesInCurrentFrame += 1;
+
+            if (automaticCaptureHz == 0.0f)
+            {
+                return;
+            }
+
+            timer += Time.deltaTime;
```

Comment:
This may be source for some errors -> you use Time.deltaTime inside FixedUpdate, where we should rather use Time.fixedDeltaTime. Were there no possible issues related to this? Or this works correctly because of something?

Similar issue may be in SetVelocityToRaytrace, depending on where do you use Capture method (Update or FixedUpdate or e.g. on demand).

I see that LidarSensor also uses Time.deltaTime inside FixedUpdateLogic, that is non standard and may be technically used in different places.

Reply:
> When this is called from inside [MonoBehaviour.FixedUpdate](https://docs.unity3d.com/ScriptReference/MonoBehaviour.FixedUpdate.html), it returns [Time.fixedDeltaTime](https://docs.unity3d.com/ScriptReference/Time-fixedDeltaTime.html).

From [Time.deltaTime Unity Docs](https://docs.unity3d.com/ScriptReference/Time-deltaTime.html)